src/sidebar/services/features.js

```diff
@@ -1,60 +1,43 @@
+import bridgeEvents from '../../shared/bridge-events';
+import { watch } from '../util/watch';
+
 /**
- * Provides access to feature flag states for the current
- * Hypothesis user.
+ * Service that provides operations related to feature flags.
+ *
+ * Feature flags information is part of the user's profile and in the sidebar
+ * is accessed via the store. This service synchronizes the state of feature
+ * flags to the `annotator` side of the application.
  *
- * This service is a thin wrapper around the feature flag data in
- * the session state.
+ * Note that the state of feature flags can change whenever the active profile
+ * information changes.
  *
- * Users of this service should assume that the value of any given flag can
- * change at any time and should write code accordingly. Feature flags should
- * not be cached, and should not be interrogated only at setup time.
+ * @inject
  */
-
-import bridgeEvents from '../../shared/bridge-events';
-import warnOnce from '../../shared/warn-once';
-import { watch } from '../util/watch';
-
-// @inject
-export default function features(bridge, session, store) {
-  const currentFlags = () => store.profile().features;
-  const sendFeatureFlags = () => {
-    bridge.call(bridgeEvents.FEATURE_FLAGS_UPDATED, currentFlags() || {});
-  };
-
-  // Re-send feature flags to connected frames when flags change or a new
-  // frame connects.
-  watch(store.subscribe, [currentFlags, () => store.frames()], () =>
-    sendFeatureFlags()
-  );
-
+export class FeaturesService {
   /**
-   * Returns true if the flag with the given name is enabled for the current
-   * user.
-   *
-   * Returns false if session data has not been fetched for the current
-   * user yet or if the feature flag name is unknown.
+   * @param {import('../../shared/bridge').default} bridge
+   * @param {import('../store').SidebarStore} store
    */
-  function flagEnabled(flag) {
-    // trigger a refresh of session data, if it has not been
-    // refetched within a cache timeout managed by the session service
-    // (see CACHE_TTL in session.js)
-    session.load();
-
-    const flags = currentFlags();
-
-    if (!flags) {
-      // features data has not yet been fetched
-      return false;
-    }
-
-    if (!(flag in flags)) {
-      warnOnce('looked up unknown feature', flag);
-      return false;
-    }
-    return flags[flag];
+  constructor(bridge, store) {
+    this._bridge = bridge;
+    this._store = store;
   }
 
-  return {
-    flagEnabled: flagEnabled,
-  };
+  init() {
+    const currentFlags = () => this._store.profile().features;
+    const sendFeatureFlags = () => {
+      this._bridge.call(
+        bridgeEvents.FEATURE_FLAGS_UPDATED,
+        currentFlags() || {}
+      );
+    };
+
+    // Re-send feature flags to connected frames when flags change or a new
+    // frame connects.
+    watch(
+      this._store.subscribe,
+      [currentFlags, () => this._store.frames()],
+      () => sendFeatureFlags()
+    );
```

Comment:
```suggestion
      [currentFlags, this._store.frames],
      sendFeatureFlags
    );
```

Reply:
Practically yes, although in the first case it makes an assumption that `frames` is a bound method. That's currently true and I don't see it changing. In general its a bit of an unfortunate refactoring hazard though.